api/routes/settings.ts

```diff
@@ -0,0 +1,70 @@
+import { FastifyPluginCallback } from 'fastify';
+import { Type } from '@sinclair/typebox';
+
+export const settingRoutes: FastifyPluginCallback = (
+  fastify,
+  _options,
+  done
+) => {
+  fastify.put<{
+    Body: {
+      profileUI: {
+        isLocked: boolean;
+        showAbout: boolean;
+        showCerts: boolean;
+        showDonation: boolean;
+        showHeatMap: boolean;
+        showLocation: boolean;
+        showName: boolean;
+        showPoints: boolean;
+        showPortfolio: boolean;
+        showTimeLine: boolean;
+      };
+    };
+  }>(
```

Comment:
```suggestion
  fastify.put(
```
This can be inferred.